goalie/go_mesh_seq.py

```diff
@@ -228,6 +232,17 @@ def indicate_errors(
                     indi.interpolate(abs(indi))
                     self.indicators[f][i][j].interpolate(ufl.max_value(indi, 1.0e-16))
 
+            # discard extra subinterval solution field
+            self.solutions[f][FWD_OLD].pop()
+            self.solutions[f][ADJ_NEXT].pop()
+            enriched_mesh_seq.solutions[f][FWD_OLD].pop()
+            enriched_mesh_seq.solutions[f][ADJ_NEXT].pop()
+
+            # delete extra mesh to reduce the memory footprint
+            enriched_mesh_seq.meshes.pop()
+            enriched_mesh_seq.time_partition.num_subintervals -= 1
+            enriched_mesh_seq.time_partition.timesteps.pop()
```

Comment:
Now that I see the code... doesn't the entire `enriched_mesh_seq` get destroyed after each call to `indicate_errors`?